src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/TagBuilder.cs

```diff
@@ -31,16 +23,11 @@ public TagBuilder(string tagName, [NotNull] IHtmlEncoder htmlEncoder)
 
             TagName = tagName;
             Attributes = new SortedDictionary<string, string>(StringComparer.OrdinalIgnoreCase);
-            _htmlEncoder = htmlEncoder;
         }
 
         public IDictionary<string, string> Attributes { get; private set; }
 
-        public string InnerHtml
-        {
-            get { return _innerHtml ?? string.Empty; }
-            set { _innerHtml = value; }
-        }
+        public IHtmlContent InnerHtml { get; set; } = HtmlString.Empty;
```

Comment:
Previously could be set to `null` without causing an NRE in later code. Probably should remain that defensive.

Other option is to add `[param: NotNull]` above the `set`. Minor compat break but no NRE.